### PR TITLE
Add ability to resize buttons

### PR DESCRIPTION
**Pull request by tomrimk**

### CODE REVIEW

File: packages/viewer/web/src/getPluginViewers.jsx [hunk 1]
```diff
@@ -142,6 +146,13 @@ class PluginViewer extends PureComponent {
             customStyles = { width: src.width, maxWidth: '100%' };
           }
         }
+
+        if (type === ACTION_BUTTON_TYPE || type === LINK_BUTTON_TYPE) {
```
Comment (talevy17) [inline packages/viewer/web/src/getPluginViewers.jsx:150]: this "customStyles" section gathers a lot of logic, seems a bit messy. 
its unrelated to this PR but maybe we should rework this mechanism, it doesn't scale well IMO.

**Review by talevy17 — APPROVED** [review]
overall seems solid, we should consider unit testing getPluginViewers - seems to gather a lot of logic.

File: packages/viewer/web/src/getPluginViewers.jsx (hunk 1 above)
Comment (maimonav) [reply packages/viewer/web/src/getPluginViewers.jsx:150]: Agree.

File: packages/plugin-button/web/src/createButtonPlugin.ts [hunk 2]
```diff
@@ -61,6 +71,30 @@ const createButtonPlugin = (
...
+      }
+      return PLUGIN_DECORATION_PROPS[PLUGIN_DECORATIONS.RESIZEABLE](calulatedProps);
+    },
+    componentWillReceiveDecorationProps: (props, nextProps, onPropsChange) => {
+      const { width } = PLUGIN_DECORATION_PROPS[PLUGIN_DECORATIONS.RESIZEABLE](props);
+      const { width: nextWidth } = PLUGIN_DECORATION_PROPS[PLUGIN_DECORATIONS.RESIZEABLE](
+        nextProps
+      );
+      if (width !== nextWidth) {
+        onPropsChange({ width: nextWidth, size: ButtonType.CUSTOM });
```
Comment (mharsat) [inline packages/plugin-button/web/src/createButtonPlugin.ts:95]: This size should be one of the known sizes right? Like in this case `inline`

Comment (maimonav) [reply packages/plugin-button/web/src/createButtonPlugin.ts:95]: It doesn't really matter, but I changed it to size `inline` for consistency.
To make it work in new schema, `inline` should have automatically convert to `CUSTOM`.

Comment (mharsat) [reply packages/plugin-button/web/src/createButtonPlugin.ts:95]: great I'll take care of that side

Comment (maimonav) [reply packages/plugin-button/web/src/createButtonPlugin.ts]: TOTACH

**Review by mharsat — APPROVED** [review]
As long as `width` and `size` are located in the config this will work in schema conversion.
Just make sure to check the conversion in the example before merging

**Comment by maimonav** [comment]
> As long as `width` and `size` are located in the config this will work in schema conversion.
> Just make sure to check the conversion in the example before merging

Cool, they are located like you described, but don't forget to convert "inline" to "CUSTOM", since right now, it converts it to "UNKNOWN".

**Pull request merged**